crates/completion/src/item.rs

```diff
@@ -310,32 +337,28 @@ impl Builder {
                 }
                 label = format!("{}::{}", import_path_without_last_segment, label);
             }
-
-            let rewriter = insert_use::insert_use(
-                &import_data.import_scope,
-                import,
-                import_data.merge_behaviour,
-            );
-            if let Some(old_ast) = rewriter.rewrite_root() {
-                algo::diff(&old_ast, &rewriter.rewrite(&old_ast)).into_text_edit(&mut text_edits);
-            }
         }
 
-        let original_edit = match self.text_edit {
+        let mut text_edit = match self.text_edit {
             Some(it) => it,
             None => {
                 TextEdit::replace(self.source_range, insert_text.unwrap_or_else(|| label.clone()))
             }
         };
 
-        let mut resulting_edit = text_edits.finish();
-        resulting_edit.union(original_edit).expect("Failed to unite text edits");
+        if !self.resolve_import_lazily {
+            if let Some(import_edit) =
+                self.import_to_add.as_ref().and_then(|import_edit| import_edit.to_text_edit())
+            {
+                text_edit.union(import_edit).expect("Failed to unite import and completion edits");
```

Comment:
I get nervous about panicing since I've seen it take down the server during assists. Is there any way we can avoid it?

Reply:
The question here is "what to do if not to panic?". 
IMO we should not return a half-baked CompletionItem in such case at all, but if I change the return type to `Option` or `Result`, the change will touch a lot of other code, so I did not dare to do this in this PR.

But sure, I've adjusted the code to log an error instead and to return some completion nontheless.
I did not remove the panic on serialisation in `handlers.rs` , since the whole file is abundant with them anyway.